Release-0.15 follow-up PR (#2674)
prometheus-operator/kube-prometheus: RELEASE.md

## Update Kubernetes supported versions

> The `main` branch of kube-prometheus should support the atleast last 2 versions of

Reviewer: ```suggestion
The `main` branch of kube-prometheus should support at least the last 2 versions of
```

Once the PR cutting the release is merged, pull the changes, create a new
release branch named `release-x.y` based on the latest changes and push it to
the upstream repository or create the branch from Github UI directly.

## Create the release

> From Github UI and draft a new [release](https://github.com/prometheus-operator/kube-prometheus/releases/new). Give the correct tag name and select the newly created release branch as Target. Fill the description and click Publish release.

Reviewer: ```suggestion
From the Github UI, draft a new [release](https://github.com/prometheus-operator/kube-prometheus/releases/new). Give the correct tag name and select the newly created release branch as Target. Fill the description and click the `Publish release` button.
```

> **Note:** New tag will be created automatically when the release is published.

Reviewer: ```suggestion
**Note:** the new tag will be created automatically when the release is published.
```
